test.py

```diff
@@ -83,10 +83,21 @@ def eval_fn(self):
             except NotImplementedError:
                 self.skipTest('Method eval is not implemented, skipping...')
 
+    def check_device_fn(self):
+        task = ModelTask(path, timeout=TIMEOUT)
+        with task.watch_cuda_memory(skip=(device != "cuda"), assert_equal=self.assertEqual):
```

Comment:
I guess this means we only run the check_device test when the device is cuda and skip the test when device is CPU?

Reply:
This test should also run for CPU. I think `watch_cuda_memory` is to check for memory leaks when running on `device == cuda`, otherwise `watch_cuda_memory` is a no-op.